modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -494,14 +496,15 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
         //Populate required experiment metadata along with all the existing user assignments for the given application
         assignmentsRepository.populateExperimentMetadata(userID, applicationName, context, experimentBatch, allowAssignmentsOptional, appPriorities, experimentMap, userAssignments, bucketMap, exclusionMap);
 
+        List<Pair<Experiment, Assignment>> assignmentPairs = new LinkedList<>();
         List<Map> allAssignments = new LinkedList<>();
         // iterate over all experiments in the application in priority order
         for (PrioritizedExperiment experiment : appPriorities.getPrioritizedExperiments()) {
-            if(LOGGER.isDebugEnabled()) LOGGER.debug("Now processing: {}", experiment);
+            LOGGER.debug("Now processing: {}", experiment);
 
             //check if the experiment was given in experimentBatch
             if (experimentBatch.getLabels().contains(experiment.getLabel())) {
-                if(LOGGER.isDebugEnabled()) LOGGER.debug("Experiment ({}) has given batch assignment....", experiment.getLabel());
+                LOGGER.debug("Experiment ({}) has given batch assignment....", experiment.getLabel());
 
                 String labelStr = experiment.getLabel().toString();
                 Map<String, Object> tempResult = new HashMap<>();
```

Comment:
This comment is for line 345 getAssignment() method. This method should not ingest messages to Ingestion system because assignment is not assigned to user or created yet. `        // Ingest data to real time data ingestion systems if executors exist
		for (String name : executors.keySet()) {
			executors.get(name).execute(new AssignmentEnvelopePayload(userID, context, createAssignment, false,
                    ignoreSamplingPercent, segmentationProfile, assignment != null ? assignment.getStatus() : null,
                    assignment != null ? assignment.getBucketLabel() : null, pageName, applicationName, experimentLabel,
                    experimentID, currentDate, headers));
		}        
`

Also not sure, it should update rule cache.

`                    // Make a call to Rule cache update
                    ruleCacheExecutor.execute(new ExperimentRuleCacheUpdateEnvelope(experiment.getRule(),
                            ruleCache, experimentID));
`